service/controller/clusterapi/v17/key/cluster_provider.go

```diff
@@ -60,3 +60,18 @@ func g8sClusterStatusFromCMAClusterStatus(cmaStatus *runtime.RawExtension) g8sv1
 
 	return g8sStatus
 }
+
+func setG8sClusterStatusToCMAClusterStatus(cluster cmav1alpha1.Cluster, status g8sv1alpha1.AWSClusterStatus) cmav1alpha1.Cluster {
```

Comment:
For me signature of this method is very confusing. If the method name start with `set` I think it should mutate the object. IMO `cluster` arg should be passed as a pointer and mutated in place. Immutability here brings only confusion here.

Reply:
Ack. I figured making this immutable as that's what we usually do, but I'm fine with in-place setting. Unless there are further objections, I'll provider a change for this.

Reply:
Alternatively `set` should be replaced with `with` i.e. `withG8sClusterStatusToCMAClusterStatus`.